h/bench_api.h

```diff
@@ -12,6 +12,9 @@
 #ifdef FREERTOS
 #include "../freertos/bench_porting_layer_freertos.h"
 #endif /* FREERTOS */
+#ifdef RTEMS
```

Comment:
I wonder if cmake can't help us here - there are two of these ifdef chains already...
Or maybe ensure bench_api has no need to leak anything from the rtos layer - why is it needed here?

Reply:
At the moment it is needed at least because of bench_time_t.

I agree that the code has an "ugly" feel to it. It is something that I would like to see cleaned up in the future.